packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/async_jobs/async_jobs.py

```diff
@@ -184,45 +186,72 @@ async def result(self) -> Any:
         return await self._result
 
 
-async def submit_and_wait(
+async def wait_and_get_result(
     rabbitmq_rpc_client: RabbitMQRPCClient,
     *,
     rpc_namespace: RPCNamespace,
     method_name: str,
+    job_id: AsyncJobId,
     job_id_data: AsyncJobNameData,
     client_timeout: datetime.timedelta,
-    **kwargs,
 ) -> AsyncGenerator[AsyncJobComposedResult, None]:
-    async_job_rpc_get = None
+    """when a job is already submitted this will wait for its completion
+    and return the composed result"""
     try:
-        async_job_rpc_get = await submit(
-            rabbitmq_rpc_client,
-            rpc_namespace=rpc_namespace,
-            method_name=method_name,
-            job_id_data=job_id_data,
-            **kwargs,
-        )
-        job_status: AsyncJobStatus | None = None
```

Comment:
Q: why you dropped the annotation? any particular reason?

Reply:
did I?